core/controllers/classroom.py

```diff
@@ -55,18 +55,16 @@ def get(self, classroom_url_fragment):
         topic_ids = classroom.topic_ids
         topic_summaries = topic_services.get_multi_topic_summaries(topic_ids)
         topic_rights = topic_services.get_multi_topic_rights(topic_ids)
-        public_topic_summary_dicts = [
-            summary.to_dict() for ind, summary in enumerate(topic_summaries)
-            if summary is not None and topic_rights[ind].topic_is_published
-        ]
-        private_topic_summary_dicts = [
-            summary.to_dict() for ind, summary in enumerate(topic_summaries)
-            if summary is not None and not topic_rights[ind].topic_is_published
-        ]
+        topic_summary_dicts = []
+        for index, summary in enumerate(topic_summaries):
+            if summary is not None:
+                topic_summary_dict = summary.to_dict()
+                topic_summary_dict['is_published'] = (
```

Comment:
Does the topic summary object in the frontend already have this field?

Reply:
Yup

https://github.com/oppia/oppia/blob/74379473dd49a2ed7becb7d1fd24ed45253105d5/core/templates/domain/topic/topic-summary.model.ts#L36-L38